data-prepper-core/src/main/java/org/opensearch/dataprepper/event/DefaultEventFactory.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.event;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventFactory;
+import org.opensearch.dataprepper.model.event.LogEventBuilder;
+import org.opensearch.dataprepper.model.event.BaseEventBuilder;
+
+public class DefaultEventFactory implements EventFactory {
+    @Override
+    public <T extends Event, B extends BaseEventBuilder<T>> B eventBuilder(Class<B> eventBuilderClass) {
+        if (eventBuilderClass.equals(LogEventBuilder.class)) {
```

Comment:
We can produce these via dependency injection and Spring's application context.

Perhaps the following structure (all within data-prepper-core):

```
abstract class DefaultEventBuilderFactory {
  abstract Class<?> getEventClass();
  abstract DefaultBaseEventBuilder createNew();
}
```

Then each event builder would look more like this:

```
@Component // puts in Spring application context
class DefaultLogEventBuilderFactory implements DefaultEventBuilderFactory {
  Class<?> getEventClass() {
    return LogEventBuilder.class;
  }
  DefaultBaseEventBuilder createNew() {
    return new DefaultLogEventBuilder();
  }

  public static class DefaultLogEventBuilder extends DefaultBaseEventBuilder<Log> implements LogEventBuilder  {
    // just as you have it now.
  }
}
```

Finally, since these are all in the application context, the `DefaultEventFactory` can get these through DI:

```
@Component
public class DefaultEventFactory implements EventFactory {
  Map<Class<?>, DefaultEventBuilderFactory> classToFactoryMap;
  public DefaultEventFactory(Collection< DefaultEventBuilderFactory> factories) {
    // convert to map
  }

  @Override
  public <T extends Event, B extends BaseEventBuilder<T>> B eventBuilder(Class<B> eventBuilderClass) {
    return classToFactoryMap.get(eventBuilderClass).createNew();  // need handle null as well
  }
```

This approach is nice because you adding a new type just requires creating the one file with both classes.

I'm happy to accept this as a follow-on PR as well.

Reply:
Will address this in the next PR